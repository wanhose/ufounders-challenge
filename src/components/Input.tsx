import { InputHTMLAttributes, memo } from 'react';
import { Component, Error, Label } from './Input.styles';

type Props = InputHTMLAttributes<HTMLInputElement> & {
  error?: boolean;
  errorMessage?: string;
  label: string;
};

const Input = ({ className, error, errorMessage, id, label, name, ...props }: Props) => (
  <div className={className}>
    <Label as="label" htmlFor={id || name}>
      {label}
    </Label>
    <Component $error={error} id={id || name} name={name} {...props} />
    <Error>{error ? errorMessage : null}</Error>
  </div>
);

export default memo(Input);

export type InputProps = Props;
