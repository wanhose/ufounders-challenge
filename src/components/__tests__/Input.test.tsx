import { render } from 'tests/utils';

import Input from '../Input';

describe('Input', () => {
  it('renders correctly', () => {
    const { container } = render(<Input error={false} errorMessage="" label="Jest" value="jest" />);

    expect(container).toMatchSnapshot('Regular');
  });

  it('error renders correctly', () => {
    const { container } = render(<Input error errorMessage="Error" label="Jest" value="jest" />);

    expect(container).toMatchSnapshot('Error');
  });
});
