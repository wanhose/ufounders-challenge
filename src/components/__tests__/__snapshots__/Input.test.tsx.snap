// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input error renders correctly: Error 1`] = `
.c1 {
  background-color: #f2f3f4;
  border: none;
  border-radius: 0.5rem;
  color: #11142d;
  font-size: 1rem;
  line-height: 150%;
  margin-top: 0.5rem;
  outline: unset;
  padding: 1rem;
  -webkit-transition: opacity linear 0.5s;
  transition: opacity linear 0.5s;
  width: 100%;
  border: 0.0625rem solid #e93256;
  box-shadow: 0 0.25rem 1.25rem rgba(255,44,85,0.2);
}

.c1:focus,
.c1:hover {
  opacity: 0.8;
}

.c1:disabled {
  cursor: not-allowed;
  opacity: 0.4;
}

.c2 {
  color: #e93256;
  display: block;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1rem;
  margin-top: 0.25rem;
  min-height: 1rem;
}

.c0 {
  color: #858b94;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1rem;
  display: block;
}

<div>
  <div>
    <label
      class="c0"
    >
      Jest
    </label>
    <input
      class="c1"
      value="jest"
    />
    <span
      class="c2"
    >
      Error
    </span>
  </div>
</div>
`;

exports[`Input renders correctly: Regular 1`] = `
.c1 {
  background-color: #f2f3f4;
  border: none;
  border-radius: 0.5rem;
  color: #11142d;
  font-size: 1rem;
  line-height: 150%;
  margin-top: 0.5rem;
  outline: unset;
  padding: 1rem;
  -webkit-transition: opacity linear 0.5s;
  transition: opacity linear 0.5s;
  width: 100%;
}

.c1:focus,
.c1:hover {
  opacity: 0.8;
}

.c1:disabled {
  cursor: not-allowed;
  opacity: 0.4;
}

.c2 {
  color: #e93256;
  display: block;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1rem;
  margin-top: 0.25rem;
  min-height: 1rem;
}

.c0 {
  color: #858b94;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1rem;
  display: block;
}

<div>
  <div>
    <label
      class="c0"
    >
      Jest
    </label>
    <input
      class="c1"
      value="jest"
    />
    <span
      class="c2"
    />
  </div>
</div>
`;
